src/components/tl/ucp/tl_ucp_context.c

```diff
@@ -477,22 +492,42 @@ ucc_status_t ucc_tl_ucp_ctx_remote_populate(ucc_tl_ucp_context_t * ctx,
     }
 
     for (i = 0; i < nsegs; i++) {
-        mmap_params.field_mask =
-            UCP_MEM_MAP_PARAM_FIELD_ADDRESS | UCP_MEM_MAP_PARAM_FIELD_LENGTH;
-        mmap_params.address = map.segments[i].address;
-        mmap_params.length  = map.segments[i].len;
-
-        status = ucp_mem_map(ctx->worker.ucp_context, &mmap_params, &mh);
-        if (UCS_OK != status) {
-            tl_error(ctx->super.super.lib,
-                     "ucp_mem_map failed with error code: %d", status);
-            ucc_status = ucs_status_to_ucc_status(status);
-            goto fail_mem_map;
+        if (map.segments[i].resource == NULL) {
+            mmap_params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                                     UCP_MEM_MAP_PARAM_FIELD_LENGTH;
+            mmap_params.address = map.segments[i].address;
+            mmap_params.length  = map.segments[i].len;
+
+            status = ucp_mem_map(ctx->worker.ucp_context, &mmap_params, &mh);
+            if (UCS_OK != status) {
+                tl_error(ctx->super.super.lib,
+                         "ucp_mem_map failed with error code: %d", status);
+                ucc_status = ucs_status_to_ucc_status(status);
+                goto fail_mem_map;
+            }
+            ctx->remote_info[i].packed_memh = NULL;
+        } else {
```

Comment:
Should this `else` have a check for the `use_xgvmi` field? If use_xgvmi was "n" then error?